matrix.go

```diff
@@ -60,16 +31,34 @@ const (
 	Skipped
 )
 
-func ExecuteTestCase(testCase TestCase) Result {
+func Execute(matrix Matrix) []Result {
+
+	cases := Collect(matrix)
+	results := ExecuteCases(cases)
+
+	return results
+}
 
-	callUrl, err := url.Parse(fmt.Sprintf("http://%v:8080/", testCase.Client))
+func ExecuteCases(cases []TestCase) []Result {
+	results := make([]Result, 0)
```

Comment:
var result []Result

Although in this case since you know the final size, do:

    result := make([]Result, 0, len(cases))